nrf_modem/doc/modem_trace.rst

```diff
@@ -0,0 +1,68 @@
+.. _modem_trace:
+
+.. _%XMODEMTRACE AT command documentation: https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/mob_termination_ctrl_status/xmodemtrace.html
+
+Modem traces
+############
+
+The modem trace interface in the Modem library is used to retrieve trace data from the modem core.
+
+Enabling modem traces
+*********************
+
+To enable traces the application must initialize the Modem library with a non-zero trace region size.
+The trace output from the modem can be toggled at runtime using the AT command ``%XMODEMTRACE``, which can also be used to change the trace output level.
+See the AT command documentation `%XMODEMTRACE AT command documentation`_ for further information.
+
+Fetching modem traces
+*********************
+
+Modem traces are retrieved by the application and output to a trace medium, typically UART.
+You can call the :c:func:`nrf_modem_trace_get` function to retrieve traces from the Modem library.
+
+The trace data, containing multiple fragments, must be processed in order, starting from the first array index.
+
+Once the trace data is processed, the application must call the :c:func:`nrf_modem_trace_processed` function to release the memory allocated in the trace memory area.
+If the trace data is processed in fragments, this can be split into multiple calls, for example, one call for each fragment of processed data.
+
+.. figure:: images/nrf_modem_trace_sequence_data_available.svg
+   :alt: Modem library trace processing
+
+   Modem library trace processing
+
+.. note::
+
+	If no trace data is available, the :c:func:`nrf_modem_os_trace_get` function blocks until new data becomes available or the modem is shut down or faulted.
+	This require the traces to be processed in its own thread separate from the rest of the application.
+
+Processing modem traces
+***********************
+
+The application must call the: c:func: `nrf_modem_trace_processed` function as it processes the trace data to inform the Modem library that the trace data has been handled and can be released.
```

Comment:
```suggestion
The application must call the: c:func:`nrf_modem_trace_processed` function as it processes the trace data to inform the Modem library that the trace data has been handled and can be released.
```

c:func:`nrf_modem_trace_processed` - Some of the modem trace functions don't link. I couldn't find the doxyegn details of the modem trace function in any of the header files. Could we also add the modem trace functions to the API documentation (doxygen details) so that they are linked accordingly?

Reply:
I guess the reason why the files don't link is because they aren't exported to nrfxlib before the release. I'll add a temporary commit with new headers here.